### ISSUE TITLE
NoMethodError: undefined method `parent' for [sequence]

### ISSUE THREAD

**Issue by adambedford** [joelmoss/sidekiq-sequence#4]
Running into this issue where the first step cannot trigger the second because `parent` doesn't exist on the worker/step class.

Depending on the module hierarchy in use, this could potentially be rectified by looking at `self.class.module_parents.first` but that would require steps to be very closely coupled to the sequence.

What do you think about storing the parent class in the DB so each step can grab that reference + steps could be shared across sequences?

Also, how is this working for you at the moment? Quite possible I'm just missing something! 😄

**Comment by joelmoss** [comment]
Hey 👋

You got a sample of the code you're using?

**Comment by adambedford** [comment]
Hey @joelmoss Thanks for your message.



Here's the sequence:

```ruby
module Sequences
  module Facebook
    class FetchHistoricalDataSequence < Sidekiq::Sequence::Base
      step FetchAdCampaignsStep
      step FetchAdSetsStep
      step FetchAdsStep
      step FetchInsightsForAdAccountStep
      step FetchInsightsStep
    end
  end
end
```

Here's the first step:
```ruby
module Sequences
  module Facebook
    class FetchHistoricalDataSequence::FetchAdCampaignsStep
      include Sidekiq::Sequence::Worker

      def perform
        facebook_ad_account_id = @data.dig(:facebook_ad_account, :id)
        facebook_ad_account = FacebookAdAccount.find(facebook_ad_account_id)
        IntegrationActions::Facebook::FetchAdCampaigns.call(service: facebook_ad_account.service)
      end
    end
  end
end
```

**Comment by adambedford** [comment]
I got things working by manually setting a class attribute on each of the steps:

```ruby
class_attribute :parent
self.parent = Sequences::Facebook::FetchHistoricalDataSequence
```

but I suspect this isn't / shouldn't be a required step?

**Comment by joelmoss** [comment]
OK, so it looks like this is a change in Rails, where `Module#parent` and the like were renamed to `Module#module_parent` in Rails 6.0, and the old removed in 6.1.

All is fixed in [0.1.10](https://github.com/joelmoss/sidekiq-sequence/releases/tag/v0.1.10)

thx for the bug report 🙏
